data-hibernate6-jpa/src/main/java/io/micronaut/data/jpa3/repository/criteria/Specification.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.jpa3.repository.criteria;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+
+/**
+ * Specification in the sense of Domain Driven Design.
+ * Fork from org.springframework.data.jpa.domain.Specification.
+ *
+ * @author Oliver Gierke
+ * @author Thomas Darimont
+ * @author Krzysztof Rzymkowski
+ * @author Sebastian Staudt
+ * @author Mark Paluch
+ * @author Jens Schauder
+ * @param <T> The entity root type
+ * @since 3.1
+ */
+public interface Specification<T> {
```

Comment:
Some of this classes should be removed and `data-model` `io.micronaut.data.repository.jpa`  should be used

Reply:
So I have tried to do this and faced number of challenges. In the end, when I deleted JpaSpecificationExecutor from io.micronaut.data.jpa3.repository then it finds interceptor FindPageSpecificationInterceptor in io.micronaut.data.runtime.intercept.criteria but fails with `Cannot determine QueryBuilder` so maybe there was a reason why this was copied here. It executes `io.micronaut.data.runtime.intercept.criteria.AbstractSpecificationInterceptor` which uses different logic than currently `io.micronaut.data.jpa3.repository.intercept.AbstractSpecificationInterceptor` and I can't find the way to change this.

Reply:
This matcher is finding it 
```
if (isFirstParameterMicronautDataQuerySpecification(matchContext.getMethodElement())) {
                Map.Entry<ClassElement, ClassElement> e = FindersUtils.pickFindPageSpecInterceptor(matchContext, matchContext.getReturnType());
                return mc -> new MethodMatchInfo(
                        DataMethod.OperationType.QUERY,
                        e.getKey(),
                        e.getValue()
                );
            }
```
so I think we should leave it